XLS-30/README.md

```diff
@@ -0,0 +1,1254 @@
+- $\Gamma_{LPTokens} = \Gamma_{LPTokens} + \Delta_{LPTokens}$: Current new balance of outstanding `LPTokens`
+
+#### 2.4.1.3 Single asset deposit
+
+Let $B$ be the only asset being deposited in return for $\Delta_{LPTokens}$, then this is executed as a combination of _equal asset_ deposit and a _swap_ transaction. The trading fee is only charged on the _amount_ of asset being swapped.
+
+Also let 
+
+- $F_{1} = (1-TFee)$
+- $F_{2} = \frac{(1 - \frac{TFee}{2})}{F_1}$, then
+
+
+$$\Delta_{LPTokens} = \Gamma_{LPTokens} * \left[\frac{\frac{\Delta_{B}}{\Gamma_{B}} - \left[\left({F_2}^{\frac{1}{W_{B}}} + \frac{\Delta_{B}}{(\Gamma_{B}*F_1)}\right)^{W_B} - F_2\right]} 
+        {1 + {\left(F_2^{\frac{1}{W_B}} + \frac{\Delta_{B}}{(\Gamma_{B}*F_1)}\right)}^{W_{B}} - F_2}\right] \tag{3}$$
+
+
+Similarly, we can derive $\Delta{B}$ from the above equation. We call this equation 4.
+
+Following is the updated pool composition of the AMM instance after successful trade:
+
+- $\Gamma_{B} = \Gamma_{B} + \Delta_{B}$: Current new balance of asset $B$
+- $\Gamma_{LPTokens} = \Gamma_{LPTokens} + \Delta_{LPTokens}$: Current new balance of outstanding `LPTokens`
+
+
+#### 2.4.1.4. Specifying different parameters
+The proposal allows for traders to specify different combinations of the above mentioned fields for **`AMMDeposit`** transaction. The implementation will determine the best possible sub operations based on trader's specifications.
+We introduce the following flags to the `AMMDeposit` transaction to identify valid parameter combinations. 
+
+| Flag Name |  Hex Value | Description |
+| ---------- | :----------------: | :-------:|
+| `tfLPToken`      | 0x00010000  |   If set, it indicates `LPTokenOut` parameter and may optionally include `Amount` and `Amount2` combination|
+| `tfSingleAsset`      | 0x00080000  |   If set, it indicates `Amount` parameter and may optionally include `LPTokenOut` parameter|
+| `tfTwoAsset`      | 0x00100000  |   If set, it indicates `Amount` and `Amount2` parameter combination and may optionally include `LPTokenOut` paramater|
+ | `tfOneAssetLPToken`      | 0x00200000 |   If set, it indicates `Amount` and `LPTokenOut` parameter combination|
+ | `tfLimitLPToken`      | 0x00400000  |   If set, it indicates `Amount` and `EPrice` parameter combination|
+ | `tfWithdrawAll`      | 0x00020000  |   If set, it indicates proportional withdrawal of all `LPTokens`|
+| `tfTwoAssetIfEmpty`| 0x00800000  |   If set, it indicates that this deposit can only be submitted on an empty state AMM (LPTokens == 0)|
+
+If `tfTwoAssetIfEmpty` is set, both amounts have to be specified  and deposited into AMM as is. It is sort of like `AMMCreate` in an empty AMM state.
+ 
+ Following are the recommended valid combinations. Other invalid combinations may result in the failure of transaction.
+
+- `LPTokenOut`, `[Amount]`, `[Amount2]`
+- `Amount`, `[LPTokenOut]`
+- `Amount` , `Amount2`, `[LPTokenOut]`
```

Comment:
Could have optional `TradingFee`, which is only valid with `tfTwoAssetIfEmpty` option.

Reply:
Added.